controlplane/kubeadm/internal/cluster.go

```diff
@@ -112,26 +112,11 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 	restConfig = rest.CopyConfig(restConfig)
 	restConfig.Timeout = 30 * time.Second
 
-	if m.Tracker == nil {
-		return nil, errors.New("Cannot get WorkloadCluster: No remote Cluster Cache")
-	}
-
 	c, err := m.Tracker.GetClient(ctx, clusterKey)
 	if err != nil {
 		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
 	}
 
-	clientConfig, err := m.Tracker.GetRESTConfig(ctx, clusterKey)
-	if err != nil {
-		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
-	}
-
-	// Make sure we use the same CA and Host as the client.
-	// Note: This has to be done to be able to communicate directly on self-hosted clusters.
-	restConfig.CAData = clientConfig.CAData
-	restConfig.CAFile = clientConfig.CAFile
-	restConfig.Host = clientConfig.Host
-
 	// Retrieves the etcd CA key Pair
```

Comment:
This only made sense with the old code. Since we moved to CCT there is no reason to call GetRESTConfig twice:
```go
func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.ObjectKey) (WorkloadCluster, error) {
	// TODO(chuckha): Inject this dependency.
	// TODO(chuckha): memoize this function. The workload client only exists as long as a reconciliation loop.
	restConfig, err := remote.RESTConfig(ctx, KubeadmControlPlaneControllerName, m.Client, clusterKey)
	if err != nil {
		return nil, err
	}
	restConfig.Timeout = 30 * time.Second

	if m.Tracker == nil {
		return nil, errors.New("Cannot get WorkloadCluster: No remote Cluster Cache")
	}

	c, err := m.Tracker.GetClient(ctx, clusterKey)
	if err != nil {
		return nil, err
	}

	clientConfig, err := m.Tracker.GetRESTConfig(ctx, clusterKey)
	if err != nil {
		return nil, err
	}

	// Make sure we use the same CA and Host as the client.
	// Note: This has to be done to be able to communicate directly on self-hosted clusters.
	restConfig.CAData = clientConfig.CAData
	restConfig.CAFile = clientConfig.CAFile
	restConfig.Host = clientConfig.Host
```